overlord/devicestate/handlers_install.go

```diff
@@ -1130,6 +1132,25 @@ func (m *DeviceManager) doInstallFinish(t *state.Task, _ *tomb.Tomb) error {
 		return err
 	}
 
+	if hasSystemSeed {
+		// get the path of the kernel snap relative to the original seed's
+		// directory. since we're not renaming the seed, this will be the same
+		// relative path as if we're working under the new seed's directory
+		kernelRelPath, err := filepath.Rel(dirs.SnapSeedDir, snapSeeds[snap.TypeKernel].Path)
+		if err != nil {
+			return err
+		}
+
+		err = bootMakeRecoverySystemBootable(systemAndSnaps.Model, seedMntDir, filepath.Join("systems", systemLabel), &boot.RecoverySystemBootableSet{
```

Comment:
Is this just to generate grubenv? Probably worth adding a comment with why this is needed.

Reply:
Yes, that is all it is doing.